service/src/main/java/com/ge/predix/acs/obligation/management/ObligationServiceImpl.java

```diff
@@ -0,0 +1,162 @@
+import javax.validation.Valid;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.ge.predix.acs.commons.exception.UntrustedIssuerException;
+import com.ge.predix.acs.model.Obligation;
+import com.ge.predix.acs.obligation.management.dao.ObligationEntity;
+import com.ge.predix.acs.obligation.management.dao.ObligationRepository;
+import com.ge.predix.acs.util.collection.ValidList;
+import com.ge.predix.acs.utils.JsonUtils;
+import com.ge.predix.acs.zone.management.dao.ZoneEntity;
+import com.ge.predix.acs.zone.resolver.ZoneResolver;
+
+/**
+ * 
+ * @author Sebastian Torres Brown
+ *
+ */
+@Service
+public class ObligationServiceImpl implements ObligationService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ObligationServiceImpl.class);
+
+    @Autowired
+    private ObligationRepository obligationRepository;
+
+    @Autowired
+    private ZoneResolver zoneResolver;
+
+    private final JsonUtils jsonUtils = new JsonUtils();
+
+    @Transactional
+    @Override
+    public void upsertObligation(@Valid final Obligation obligation) {
+        try {
+            ZoneEntity zone = this.zoneResolver.getZoneEntityOrFail();
+            upsertObligation(zone, obligation);
+        } catch (Exception e) {
+            handleException(e, obligation.getName());
+        }
+    }
+
+    @Transactional
+    @Override
+    public void upsertObligations(final List<Obligation> obligations) {
+        ValidList<Obligation> validObligations = new ValidList<Obligation>(obligations);
+        upsertObligations(validObligations);
+    }
+
+    private void upsertObligations(@Valid final ValidList<Obligation> obligations) {
+        String obligationName = "";
+        try {
+            ZoneEntity zone = this.zoneResolver.getZoneEntityOrFail();
+            for (Obligation obligation : obligations) {
+                obligationName = obligation.getName();
+                upsertObligation(zone, obligation);
+            }
+        } catch (Exception e) {
+            handleException(e, obligationName);
+        }
+    }
+
+    @Override
+    public List<Obligation> getObligations() {
+        ZoneEntity zone = this.zoneResolver.getZoneEntityOrFail();
+        ArrayList<Obligation> result = new ArrayList<>();
+        List<ObligationEntity> obligationEnityList = this.obligationRepository.findByZone(zone);
+        for (ObligationEntity obligationEntity : obligationEnityList) {
+            result.add(this.jsonUtils.deserialize(obligationEntity.getObligationJson(), Obligation.class));
+        }
+        return result;
+    }
+
+    @Override
+    public Obligation getObligation(final String obligationName) {
+        ZoneEntity zone = this.zoneResolver.getZoneEntityOrFail();
+        ObligationEntity obligationEntity = this.obligationRepository.getByZoneAndObligationId(zone, obligationName);
```

Comment:
change getByZoneAndObligationId to getByZoneAndObligationName , since looks like you are looking up based on name